tutorials/integration.qmd

```diff
@@ -0,0 +1,171 @@
+Any other suitable atlas could be used, not just the one defined by normal coordinates, though each one requires different volume density corrections due to the Jacobian determinant of the pushforward.
+`Manifolds.jl` provides the function ``[`volume_density`](@ref)``{=commonmark} that calculates that quantity, denoted $\theta_p(X)$.
+See for example [^leBrigantPuechmorel2019], Definition 11, for a precise description using Jacobi fields.
+
+While many sources define volume density as a function of two points, `Manifolds.jl` decided to use the more general point-tangent vector formulation. The two-points variant can be implemented as
+```{julia}
+using Manifolds
+volume_density_two_points(M::AbstractManifold, p, q) = volume_density(M, p, log(M, p, q))
+```
+
+The simplest way to of integrating a function on a compact manifold is through a Monte Carlo integrator.
+A simple variant can be implemented as follows (assuming uniform distribution of `rand`):
+
+```{julia}
+using LinearAlgebra, Distributions, SpecialFunctions
+function simple_mc_integrate(M::AbstractManifold, f; N::Int = 1000)
+    V = manifold_volume(M)
+    sum = 0.0
+    q = rand(M)
+    for i in 1:N
+        sum += f(M, q)
+        rand!(M, q)
+    end
+    return V * sum/N
+end
+```
+
+``[`manifold_volume`](@ref)``{=commonmark} and ``[`volume_density`](@ref)``{=commonmark} are closely related to each other, though very few sources explore this connection, and some even claiming a certain level of arbitrariness in defining `manifold_volume`.
+Volume is sometimes considered arbitrary because Riemannian metrics on some spaces like the manifold of rotations are defined with arbitrary constants.
+However, once a constant is picked (and it must be picked before any useful computation can be performed), all geometric operations must follow in a consistent way: inner products, exponential and logarithmic maps, volume densities, etc.
+`Manifolds.jl` consistently picks such constants and provides a unified framework, though it sometimes results in picking a different constant than what is the most popular in some sub-communities.
+
+On Lie groups the situation is more complicated. Invariance under left or right group action is a desired property that leads one to consider Haar measures [^Tornier2020].
+It is, however, unclear what are the practical benefits of considering Haar measures over the Lebesgue measure of the underlying manifold, which often turns out to be invariant anyway.
+
+Last point before we turn to consider an example is the matter of integration through charts.
+This is an approach not currently supported by `Manifolds.jl`.
+One has to define a suitable set of disjoint charts covering the entire manifold and use a method for multivariate Euclidean integration.
+Note that ranges of parameters have to be adjusted for each manifold and scaling based on the metric needs to be applied.
+See [^BoyaSudarshanTilma2003] for some considerations on symmetric spaces.
+
```

Comment:
For me the block from the last code block until here is a lot of technical detail what manifolds does and does not provide, that I do not completely unterstand. Could we maybe move this further down so that the tutorial first illustrates how Integration can be used (i.e. the distributions section) before discussion these details?

Reply:
Yes, that is a bit too technical. I've moved that paragraph to a new section for technical details.